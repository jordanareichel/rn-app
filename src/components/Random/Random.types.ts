export type RandomProps = {
  minValue: number;
  maxValue: number;
};
