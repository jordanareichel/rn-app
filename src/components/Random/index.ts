export * from './Random.comp';
export * from './Random.types';
