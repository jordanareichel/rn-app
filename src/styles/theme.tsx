export const theme = {
  font: {
    sm: '12',
    md: '14',
    lg: '18',
  },
  colors: {
    dark: '#000000',
    light: '#ffffff',
  },
};
