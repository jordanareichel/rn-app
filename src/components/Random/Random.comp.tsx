import {Text} from '@components/Text';
import {theme} from '@styles/theme';
import _ from '@utils/object';
import React from 'react';
import {RandomProps} from './Random.types';

export const Random: React.FC<RandomProps> = props => {
  const {minValue, maxValue} = props;

  const random = Math.floor(
    Math.random() * (maxValue - minValue + 1) + minValue,
  );

  return (
    <Text
      fontSize={_.get(theme, 'font.md')}
      color={_.get(theme, 'colors.dark')}
      fontWeight={'bold'}>
      Valor aleatório é {random}
    </Text>
  );
};
