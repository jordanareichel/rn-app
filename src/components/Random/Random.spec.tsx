import React from 'react';

import TestRenderer from 'react-test-renderer';
import {ThemeProvider} from 'styled-components/native';
import {Random} from './Random.comp';
import {theme} from '@styles/theme';

const RandomComp = props => {
  const minValue = 10;
  const maxValue = 12;

  return <Random {...props} maxValue={maxValue} minValue={minValue} />;
};

describe('Component Random', () => {
  test('renderer currently', () => {
    const tree = TestRenderer.create(
      <ThemeProvider theme={theme}>
        <RandomComp />
      </ThemeProvider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
