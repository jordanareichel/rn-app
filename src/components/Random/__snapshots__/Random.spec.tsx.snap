// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Random renderer currently 1`] = `
<Text
  color="#000000"
  fontSize="14"
  fontWeight="bold"
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 14,
        "fontWeight": "bold",
        "textAlign": "center",
      },
    ]
  }
>
  Valor aleatório é 
  12
</Text>
`;
